specs/enhance_gluster_cluster_sync_logic.adoc

```diff
@@ -0,0 +1,375 @@
+```
+
+* Add additional field `client_count` under `Volume` and `Brick` classes to hold
+client connection count values
+
+* Add an additional field `time_left` under `RebalanceDetails` object to capture
+remaining time for rebalance
+
+* Add additional field `rebal_estimated_time` under `Volume` object to capture
+the estimated time to complete rebalance on the volume
+
+* Add additional field `is_arbiter` under `Brick` object to mark an arbiter
+brick
+
+* Add an additional object `GeoRepSessions` to hold the geo-rep session details
+for gluster volumes
+
+```
+TBD
+```
+
+* Add an additional field `client_connections` under `Volume` object to hold
+the client connections details
+
+* Add ad additional fields `client_connections` and `arbiter` under `Brick`
+object to hold the client connections and to mark if the brick is arbiter brick
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* Provide an API to list snapshot details for volumes
+
+GET clusters/{id}/Volumes/{volume-id}/GetVolumeSnapshots
+
+```
+[
+  {
+    'name': 'snap-1_GMT-2017.07.18-07.29.59',
+    'id': '461910d0-efe8-4c47-9b81-533bb18f8ce4',
+    'time': '2017-07-18 07:29:59',
+    'description': ''snapshot-1,
+    'status': 'in_use'
+  },
+  {
+    'name': 'snap-2_GMT-2017.07.18-07.30.15',
+    'id': '1b232c45-77c4-4101-ad12-973a410dbe20',
+    'time': '2017-07-18 07:30:15',
+    'description': ''snapshot-2,
+    'status': 'in_use'
+  }
+]
+```
+
+* Provide an API to list geo-rep session details
+
+GET clusters/{id}/Volumes/{volume-id}/GetGeoRepSessions
+
+```
+TBD
+```
+
+==== Notifications/Monitoring impact:
+None
+
+==== Tendrl/common impact:
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+All the changes above needs to be done under `gluster-inetgration` module
+
+==== Tendrl Dashboard impact:
+
+No direct impact on dashboard. Using APIs dahsboard should siplay the details in
+UI.
+
+=== Security impact:
+
+None.
+
+=== Other end user impact:
+
+User gets a set of APIs for listing volume snapshot and geo-rep sessions details
+
+=== Performance impact:
+
+None.
+
+=== Other deployer impact:
+
+None.
+
+=== Developer impact:
+
+None.
+
+
+== Implementation:
+
+* https://github.com/Tendrl/gluster-integration/issues/334
+* https://github.com/Tendrl/gluster-integration/issues/335
+* https://github.com/Tendrl/gluster-integration/issues/336
+* https://github.com/Tendrl/gluster-integration/issues/338
+* https://github.com/Tendrl/gluster-integration/issues/339
+* https://github.com/Tendrl/gluster-integration/issues/340
+* https://github.com/Tendrl/gluster-integration/issues/341
+* https://github.com/Tendrl/gluster-integration/issues/343
+* https://github.com/Tendrl/gluster-integration/issues/344
+* https://github.com/Tendrl/gluster-integration/issues/345
+
+=== Assignee(s):
+
+Primary assignee:
+  shtripat
+  nnDarshan
+
+=== Work Items:
+
+* https://github.com/Tendrl/specifications/issues/167
+
+
+== Dependencies:
```

Comment:
I believe there are some changes to the `get-state` that need to be made in gluster upstream. All the gluster upstream issues tracking these changes need to be listed as dependencies. @atinmu may be able to help with the list of these issues.